rclcpp_components/doc/architecture.rst

```diff
@@ -0,0 +1,263 @@
+Architecture of ``rclcpp_components``
+=====================================
+
+``rclcpp_components`` provides tools and defines conventions for building ``rclcpp`` nodes into dynamically loadable components.
+Building composable nodes makes it easier to compose them with other composable nodes at runtime.
+For example, it allows you to build your nodes one time but then at runtime decide whether or not to run them in their own process or combine them into a single process.
+
+It also provides some generic tools for working with composable nodes, including a composable node container, which can load and execute multiple nodes in a variety of scenarios.
+
+Terminology
+-----------
+
+- Composable Node:
+  - An ``rclcpp`` based node that is setup so that it can be composed into a single process with other composable nodes at either runtime or programming time.
+- Node Component:
+  - Same thing as Composable Node, can be used interchangeably.
+- Composable Node Container:
+  - A program that can load and host composable nodes, it is often general purpose and has not a priori knowledge about the nodes it is instantiating and executing.
+- pluginlib:
+  - ROS package that provides mechanisms to create and register plugins based on user defined interface classes, and then dynamically lookup and load plugins at runtime.
+- Stand-alone Node Executable:
+  - Convenience executable that can be used to run a composable node as a single process or have it loaded dynamically in a remote composable node container.
```

Comment:
I don't really understand the latter part of this sentence (I'm not sure what a "remote composable node container" is).  Could you maybe explain a little further or reword it?